docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,170 @@
+# Sonobuoy Tarball Client
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Objectives](#objectives)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+* [Unresolved Questions](#unresolved-questions)
+
+## Summary
+
+Sonobuoy delivers results in a gzipped tar archive. The structure of the tar
+is [documented here][tar structure] but programmatic access is up to every
+consumer to implement. The structure of the tarball can change between
+versions and consumer implementations become fragile.
+
+This proposal aims to describe an API for a client that can consumers can use
+to extract data from a Sonobuoy tarball.
+
+## Objectives
+
+Provide a friendly API for easy access to data found in Sonobuoy Tarballs
+across versions. Users should already be fairly familiar with the API below
+as it might look a lot like client-go. Instead of querying an API server we
+are querying a Sonobuoy tarball.
+
+### Goals
+
+ - Define and implement an API to access files in a Sonobuoy tarball.
+ - Support all tarball versions starting at v0.9.0.
+ - Have a familiar API.
+
+### Non-Goals
+
+- We will be unable to customize the tarball for every plugin as people can
+write custom plugins. However, there should be an easy way for folks to get
+their results using this client. The main feature we are supporting is access
+to the core data generated by the [Sonobuoy queries][queries].
+- There will be no support for tarballs before Sonobuoy version 0.9.0.
+
+## Proposal
+
+```go
+type Tarball interface {
+    // Sonobuoy dumps stuff out that can be queried throught the k8s interface.
+    // The source will be a tarball instead of a k8s server.
+    Resources() kubernetes.Interface
```

Comment:
[We save quite a bit of the API][queries]. It's not everything and I'm starting to wonder if there is a better way to extract things we care about using a dynamic client instead of hardcoded a set of things we extract. But this may be a question for later.

What do you mean by support selectors? We do allow sonobuoy to be configured to [only extract things that match certain labels or in specific namespaces][filters]

[queries]: https://github.com/heptio/sonobuoy/blob/master/pkg/discovery/queries.go#L114
[filters]: https://github.com/heptio/sonobuoy/blob/master/pkg/config/config.go#L85

Reply:
I meant that the Corev1 API lets you do things like query labels by selectors. So would something like `Resources().Nodes().List(metav1.ListOptions{LabelSelector:"k8s-app=kubernetes-dashboard"})` work?

Reply:
That's up to the implementation of the interface. I was thinking "yes" when I wrote this doc, but I'm thinking of nixing the entire Resources bit at this point. It provides very little value and will be very hard to get right.